drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -137,6 +140,14 @@ static void nrf5_rx_thread(void *arg1, void *arg2, void *arg3)
 
 /* Radio device API */
 
+static enum ieee802154_hw_caps nrf5_get_capabilities(struct device *dev)
+{
+	return IEEE802154_HW_FCS |
+		IEEE802154_HW_2_4_GHZ |
+		IEEE802154_HW_FILTER;
+}
```

Comment:
nRF 802.15.4 driver supports automatic ACKs.
It also support promiscuous mode (I'm not sure if promiscuous was tested in version copied to Zephyr repo - it is pretty old but @KaSroka should update it soon).

Reply:
For ack, see last patch of this commit. @KaSroka patch then will need rework on top of that PR.

Promiscuous mode is supported by most chip, it will "just" need  to be implemented on driver side: thus why I did not made it exposed by any yet. (And there is no net_mgmt hooks yet to enter that mode anyway)